src/ussd_public.js

```diff
@@ -312,37 +312,55 @@ go.app = function() {
         };
 
         self.send_compliment_instructions = function() {
+            var lang = engage.LANG_MAP[self.im.user.answers.state_language];
```

Comment:
In order to get to the compliment/complaint screens, the user doesn't go through the `state_language` screen, so this won't be set.

If you look at the `state_start`, if they're already registered, we use `self.im.user.set_lang` to set the language on `self.im.user.lang`, so that's a better place to find the user's language.